lib/rust/ensogl/component/slider/src/model.rs

```diff
@@ -0,0 +1,168 @@
+
+    pub root: display::object::Instance,
+    pub app:  Application,
+}
+
+impl Model {
+    pub fn new(app: &Application) -> Self {
+        let root = display::object::Instance::new();
+        let label = app.new_view::<text::Text>();
+        let value_left = app.new_view::<text::Text>();
+        let value_dot = app.new_view::<text::Text>();
+        let value_right = app.new_view::<text::Text>();
+        let background = background::View::new();
+        let track = track::View::new();
+
+        let app = app.clone_ref();
+        let scene = &app.display.default_scene;
+
+        root.add_child(&background);
+        root.add_child(&track);
+        root.add_child(&label);
+        root.add_child(&value_left);
+        root.add_child(&value_dot);
+        root.add_child(&value_right);
+
+        value_left.add_to_scene_layer(&scene.layers.label);
+        value_dot.add_to_scene_layer(&scene.layers.label);
+        value_right.add_to_scene_layer(&scene.layers.label);
+        label.add_to_scene_layer(&scene.layers.label);
+
+        let model =
+            Self { background, track, label, value_left, value_dot, value_right, root, app };
+
+        model.set_size(Vector2::new(200.0, 50.0));
+        model.value_dot.set_content(".");
+
+        model
+    }
+
+    /// Set component size
+    pub fn set_size(&self, size: Vector2<f32>) {
+        self.background.size.set(size);
+        self.track.size.set(size);
+    }
+
+    /// Set slider track color
+    pub fn set_track_color(&self, color: color::Lcha) {
+        self.track.color.set(color::Rgba::from(color).into());
+    }
+    /// Set slider background color
+    pub fn set_background_color(&self, color: color::Lcha) {
+        self.background.color.set(color::Rgba::from(color).into());
+    }
+
+    /// Set slider label visibility
+    pub fn set_label_visible(&self, enabled: bool) {
+        if enabled {
+            self.root.add_child(&self.label);
+        } else {
+            self.root.remove_child(&self.label);
+        }
+    }
+
+    /// Set visibility of value text right of the decimal point
+    pub fn set_value_decimal_visible(&self, enabled: bool) {
+        if enabled {
+            self.root.add_child(&self.value_dot);
+            self.root.add_child(&self.value_right);
+        } else {
+            self.root.remove_child(&self.value_dot);
+            self.root.remove_child(&self.value_right);
+        }
+    }
+}
+
+impl display::Object for Model {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.root
+    }
+}
+
+
+
+// ========================
+// === Shape definition ===
+// ========================
+
+struct Background {
+    pub width:  Var<Pixels>,
+    pub height: Var<Pixels>,
+    pub shape:  AnyShape,
+}
+
+impl Background {
+    /// Create new rounded background shape
+    fn new() -> Self {
+        let width: Var<Pixels> = "input_size.x".into();
+        let height: Var<Pixels> = "input_size.y".into();
+
```

Comment:
We should always leave a few pixels outside of the shape - when zooming out, these pixels will be used for anti-aliasing. We might do it automatically in the future, but now, you need to do it manually. create a constant for it, set it to 4 pixels and keep the sprite size bigger leaving 4 pixels on each side empty.